org.opent2t.sample.hub.superpopular/com.contosothings.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,220 @@
+    */
+    putDeviceDetailsAsync(deviceType, deviceId, putPayload) {
+        putPayload.id = this._hubId;
+        // build request path and body
+        var requestPath = '/api/hubsApi/';
+        var putPayloadString = JSON.stringify(putPayload);
+
+        // Make the async request
+        return this._makeRequest(requestPath, 'POST', putPayloadString);
+    }
+    
+    /**
+     * Subscribes, not implemented.
+     * 
+     * @param {string} deviceType - Device Type (e.g. 'thermostats')
+     * @param {string|number} deviceId - Id for the specific device
+     * @param {string} callbackUrl - Callback url for feed postbacks
+     * @param {HttpRequest} verificationRequest - GET request received by the server at a previously subscribed
+     *      callback URL.  This completes the verification half of the subscription.
+     * @param {HttpResponse} verificationResponseContent - Content that should be provided in the response to the
+     *      verification request as {'Content-Type': 'text/plain'}.  All responses should use code 200 unless an
+     *      error is caught.
+     * @returns {number} Object containing the subscription expiration time, and any content that
+     *      needs to be included in a response to the original request.
+     */
+    _subscribe(deviceType, deviceId, callbackUrl, verificationRequest) {
+        // fix lint error
+        console.log(deviceType + deviceId + callbackUrl + verificationRequest);
+        throw new Error("Not implemented");
+    }
+
+    /**
+     * Unsubscribes, not implemented.
+     * 
+     * @param {string} deviceType - Device Type (e.g. 'thermostats')
+     * @param {string|number} deviceId - Id for the specific device
+     * @param {string} callbackUrl - URL that will be unsubscribed
+     * @returns {request} Promise that supplies the server response
+     * 
+     */
+    _unsubscribe(deviceType, deviceId, callbackUrl) {
+        // fix lint error
+        console.log(deviceType + deviceId + callbackUrl);
+        throw new Error("Not implemented");
+    }
+
+    /**
+     * Gets all subscriptions for a device, not implemented.
+     * 
+     * @param {string} deviceType - Device Type (e.g. 'thermostats')
+     * @param {string|number} deviceId - Id for the specific device
+     * @returns {request} Promise that supplies the list of subscriptions
+     * 
+     * This is just a helper for tests and verification.
+     */
+    _getSubscriptions(deviceType, deviceId) {
+        // fix lint error
+        console.log(deviceType + deviceId);
+        throw new Error("Not implemented");
+    }
+
+    /** 
+     * Given the hub specific device, returns the opent2t schema and translator
+    */
+    _getOpent2tInfo(ctDevice) {
+        // if (ctDevice.ThingsType == 3) {
+        //     return { 
+        //         "schema": 'opent2t.p.thermostat',
+        //         "translator": "opent2t-translator-com-contosothings-thermostat"
+        //     };
+        // }
```

Comment:
thermostat is coming in next, this will be uncommented shortliy

---
In reply to: [103513348](https://github.com/openT2T/translators/pull/175#discussion_r103513348) [](ancestors = 103513348)